app/controllers/credentials_controller.rb

```diff
@@ -36,8 +36,11 @@ class CredentialsController < ApplicationController
   def update_password
     password = request.body.read
 
-    @role.credentials.password = password
-    @role.credentials.save(raise_on_save_failure: true)
+    Credentials::ChangePassword.new.(
```

Comment:
I've moved from `.(` to `.call` for invoking procs for clarity.  It's too easy to misread the shorter syntax.  IIRC ruby style guide may advocate this is as well.